### PR TITLE
Add ability to open/copy selected files from Search-results, not always ALL files

### LINKED ISSUE #15741: [Feature request] Limit scope of 'Search results' right-click 'Open all' and 'Copy Pathname(s)' commands
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description of the Issue

Scenario:  User has run many file-level searches in his Notepad++ working period, a lot of which are no longer relevant to what he is currently doing.  His _Search results_ panel is populated with much data, but he doesn't want to clear it.  He runs a new search and from this he wants to open all of the files with hits from _only_ that search.  He has no way to achieve this as the right-click command _Open all_ will, as it implies, open _all_ files in _all_ previous file-level searches ever conducted.

### Describe the solution you'd like.

User should be able to command that only file(s) from a specific previous search be opened.  The user would right-click on a "top-level" line from a prior search, i.e., the line that begins with `Search "..."`, choose e.g. _Open selected_ (the text of which would replace the current _Open all_), and then only the files with hits from that search would be opened into Notepad++.

If the user wanted the current (8.7) behavior, he could Ctrl+a (select-all) in _Search results_ before right-clicking and choosing _Open selected_.

---

Notes:

* The description discusses opening files, but a similar concept can be applied to _Copy Pathname(s)_ -- menu text on this one would change to _Copy Selected Pathname(s)_
* The overall logistical operation would be very similar to how _Copy Selected Line(s)_ currently works, see [UM](https://npp-user-manual.org/docs/searching/#kbdrightclickkbd-commands-in-the-client-area-of-a-search-results-windows-tab) for that behavior

### Debug Information

```shell
Notepad++ v8.7   (64-bit)
Build time : Sep 17 2024 - 17:06:31
Path : C:\NotepadPlusPlus\Misc\_NoBackup\Releases\npp.8.7\npp.8.7.portable.x64\notepad++.exe
Command Line : -titleAdd="8.7 64-bit" -multiInst
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
Periodic Backup : ON
OS Name : Windows 11 Pro (64-bit)
OS Version : 23H2
OS Build : 22631.4317
Current ANSI codepage : 1252
Plugins : 
    mimeTools (3.1)
    NppConverter (4.6)
    NppExport (0.4)
    PythonScript (2)
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by alankilborn**
Fix #15741

@donho You didn't get back to replying to the issue to say if you liked the idea or not, but there was not a lot of code to this so I just went ahead with it.

Note that changes to `ScintillaEditView.*` are simply to change two `getLine` functions into `const` functions (which they should have been even without this issue).  Note that the `const` is required for the code changes of this PR to build without error.

### CODE REVIEW

**Review by donho — CHANGES_REQUESTED** [review]
Please consider the remark if you have no better suggestion.

File: PowerEditor/installer/nativeLang/english_customizable.xml [hunk 1]
```diff
@@ -1718,9 +1718,9 @@ Find in all files but exclude all folders log or logs recursively:
 			<finder-collapse-all value="Fold all"/>
 			<finder-uncollapse-all value="Unfold all"/>
 			<finder-copy value="Copy Selected Line(s)"/>
-			<finder-copy-paths value="Copy Pathname(s)"/>
+			<finder-copy-paths value="Copy Selected Pathname(s)"/>
 			<finder-clear-all value="Clear all"/>
-			<finder-open-all value="Open all"/>
+			<finder-open-all value="Open Selected Pathname(s)"/>
```
[the remark this answers is not in the document]
Comment (alankilborn) [reply PowerEditor/installer/nativeLang/english_customizable.xml:1723]: I think it is OK either way, so I like the shorter one slightly better.

As (I think) English is a 100% language for @pryrt, maybe he has an opinion here?

[the remark this answers is not in the document]
Comment (pryrt) [reply PowerEditor/installer/nativeLang/english_customizable.xml:1723]: I have a slight preference for the shorter as well, but both communicate the action clearly to me.

File: PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp [hunk 2]
```diff
@@ -682,24 +682,45 @@ void Finder::deleteResult()
 	assert(size_t(_scintView.execute(SCI_GETLINECOUNT)) == _pMainFoundInfos->size() + 1);
 }
 
-vector<wstring> Finder::getResultFilePaths() const
+vector<wstring> Finder::getResultFilePaths(bool onlyInSelectedText) const
```
[the remark this answers is not in the document]
Comment (alankilborn) [reply PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:685]: `getResultFilePaths()` is called 3 times, not 2.
The 3rd call (using default argument of `false`) is in code that didn't change so it doesn't show up in the diffs from this PR.

That 3rd call is when user executes this:
image
For this case I chose to retain the behavior of searching all (not selected) pathnames.
But maybe this case should also be selected?

[the remark this answers is not in the document]
Comment (donho) [reply PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:685]: OK, please keep the name and argument.
OTOH, please remove the default value and call it always with value:
`vector<wstring> fileNames = findInFolderInfo->_pSourceFinder->getResultFilePaths(false);`

[the remark this answers is not in the document]
Comment (alankilborn) [reply PowerEditor/src/ScintillaComponent/FindReplaceDlg.cpp:685]: OK, changed in newest commit.

**Review by donho — APPROVED** [review]

**Pull request closed without merging**